File: BridgeToTurbin3/airdrop/airdrop.ts
import { Connection, Keypair, LAMPORTS_PER_SOL } from "@solana/web3.js"
import wallet from "./dev-wallet.json"

const keypair = Keypair.fromSecretKey(new Uint8Array(wallet));

// Create a Solana devnet connection object
const connection = new Connection("https://api.devnet.solana.com");

(async () => {
    try {
        // Claim 2 devnet SOL tokens (returns the transfer hash)
        const txhash = await connection.requestAirdrop(keypair.publicKey, 2 * LAMPORTS_PER_SOL);
        console.log(`Success! Check out your TX here: 
        https://explorer.solana.com/tx/${txhash}?cluster=devnet`);
    } catch(e) {
        console.error(`Oops, something went wrong: ${e}`)
    }
})();

// terminal -> yarn airdrop

File: rs/src/rust_registration/airdrop.rs
use solana_sdk::signature::{read_keypair_file, Keypair, Signer};
use solana_client::rpc_client::RpcClient;

const RPC_URL: &str = "https://api.devnet.solana.com";

pub fn airdrop() {
    // Import our keypair
    let keypair = read_keypair_file("dev-wallet.json").expect("Couldn't find wallet file");

    let client = RpcClient::new(RPC_URL);

    match client.request_airdrop(&keypair.pubkey(), 2_000_000_000u64) {
        Ok(s) => {
        println!("Success! Check out your TX here:");
        println!("https://explorer.solana.com/tx/{}?cluster=devnet", s.to_string());
        },
        Err(e) => println!("Oops, something went wrong: {}", e.to_string())
        };
}



// region : Note to Self
/*

 Import RpcClient to establish a connection to the Solana devnet. To import RpcClient, first add solana-client to Cargo.toml then "cargo build"
 

 */
//endregion : Note to Self